vizro-core/src/vizro/models/_controls/filter.py

```diff
@@ -335,3 +335,28 @@ def _get_options(
         # changes. See https://pandas.pydata.org/docs/whatsnew/v2.1.0.html#whatsnew-210-enhancements-new-stack.
         targeted_data = pd.concat([targeted_data, pd.Series(current_value)]).stack().dropna()  # noqa: PD013
         return sorted(set(targeted_data) - {ALL_OPTION})
+
+    def _get_proposed_targets(self, page):
+        """Get all valid figure model targets for this control based on its location in the page hierarchy."""
+        if self.targets:
+            return self.targets
+
+        parent_container = self._find_parent_container(page)
+        target_scope = parent_container if parent_container else page
+
+        return [
+            model.id for model in cast(Iterable[VizroBaseModel], model_manager._get_models(FIGURE_MODELS, target_scope))
+        ]
+
+    def _find_parent_container(self, page):
+        """Find the container that contains this control, if any."""
+        page_containers = cast(Iterable[Container], model_manager._get_models(Container, page))
```

Comment:
Do we actually need to cast?
```suggestion
        page_containers =model_manager._get_models(Container, page)
```